File: app/src/main/java/ru/voxp/android/di/presentation/ViewModelKey.java
package ru.voxp.android.di.presentation;

import dagger.MapKey;
import ru.jewelline.mvvm.interfaces.presentation.ViewModel;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@MapKey
@Target({ElementType.METHOD})
@Retention(RetentionPolicy.SOURCE)
@interface ViewModelKey {
    Class<? extends ViewModel> value();
}

File: 04_voxp_app/src/main/java/ru/voxp/android/presentation/di/ViewModelKey.kt
package ru.voxp.android.presentation.di

import dagger.MapKey
import ru.jewelline.mvvm.interfaces.presentation.ViewModel
import kotlin.annotation.AnnotationRetention.SOURCE
import kotlin.reflect.KClass

@MapKey
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(SOURCE)
internal annotation class ViewModelKey(val value: KClass<out ViewModel<*>>)